packages/block-ui/src/BlockUI.js

```diff
@@ -0,0 +1,185 @@
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import DefaultLoader from './Loader';
+import safeActiveElement from './safeActiveElement';
+
+const BlockUi = ({
+  tag: Tag,
+  blocking,
+  className,
+  children,
+  message,
+  loader: Loader,
+  renderChildren,
+  keepInView,
+  ...attributes
+}) => {
+  const [top, setTop] = useState('50%');
+  const helper = useRef();
+  const blocker = useRef();
+  const topFocus = useRef();
+  const container = useRef();
+  const messageContainer = useRef();
+
+  const keepInViewFunc = () => {
```

Comment:
I think because this gets defined every render, it will cause a memory leak when attaching and attempting to detach the listener. My thought is that this function is created and then attached by reference. Later, when attempting to detach, `keepInViewFunc` is no longer the same reference as the one that was attached and thus the `window.removeEventListener('scroll', keepInViewFunc);` doesn't find an existing listen with the new `keepInViewFunc`.

`useRef` may be able to solve this as the ref(erence) can be a function (or any referencable thing (such as an object)) and not just a react ref.

Reply:
I think I am just going to put the declaration of the function inside of the effect.

I could possibly just use the `useCallback` hook that will create a memoized function.